### DISCUSSION TITLE
CORS issue

### DISCUSSION THREAD

**Discussion by jeff-h** [plausible/analytics#1015] [Self-Hosted Support] [answered]
I've got Plausible running at https://plausible.mydomain.com and want to track a few sites eg:

* sub.mydomain.com
* anotherdomain.com

I have everything set up however the embedded script is failing with a CORS error:

![image](https://user-images.githubusercontent.com/841871/117756808-f6b4ec00-b272-11eb-9742-61ba46de150b.png)

The developer tools console has this error:

Access to XMLHttpRequest at 'https://plausible.mydomain.com/api/event' from origin 'https://sub.mydomain.com' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.

Hoping someone can help!

**Comment by jeff-h** [comment, marked as answer]
I got this working in the end, but I'm a bit confused as to why I was getting the error above. The reason things wouldn't work for me was that I had an nginx reverse proxy with basic auth in front of plausible, and the basic auth was stopping the embedded Plausible JS file from loading. Once I removed the basic auth, Plausible worked just fine.

**Reply by metmarkosaric** [reply]
thanks for the update!

**Reply by krchavdarov** [reply]
hello @jeff-h,
whats mean " Once I removed the basic auth "
i have same issue and my head is going to explode with nerves

**Reply by jeff-h** [reply]
hey @krchavdarov — hang in there, Plausible is worth it :) Basic Auth is where the browser asks for your username & password, rather than the website it's serving. You might recognise it better from [pictures](https://www.google.com/search?as_st=y&tbm=isch&as_q=basic+auth+browser+popup&as_epq=&as_oq=&as_eq=&imgsz=&imgar=&imgc=&imgcolor=&imgtype=&cr=&as_sitesearch=&safe=images&as_filetype=&tbs=).

Do you have this kind of basic auth configured?

**Reply by krchavdarov** [reply]
no, i have no auth boxes in my site
once i get this error 
![Screenshot_31](https://user-images.githubusercontent.com/25974442/180726564-0cf22866-d339-4610-b337-84a562fe1940.png)
and sometimes this
![Screenshot_32](https://user-images.githubusercontent.com/25974442/180726505-405543e5-877e-4450-bfab-b650f589e4bb.png)

**Reply by jeff-h** [reply]
Can you tell me how you're hosting Plausible? With the provided Docker Compose config? With an Nginx reverse proxy perhaps?

**Reply by krchavdarov** [reply]
With the provided Docker Compose config and I additionally installed nginx

**Reply by jeff-h** [reply]
If you don't have basic auth configured in Nginx, then unfortunately your CORS issue is caused by something different and my solution earlier in this thread won't help you. 

In your browser's developer tools, you can inspect the headers (in both the request and the response). You should see references there to `access-control-allow-origin`. This should give you some clues as to why the request is being blocked. Is there anything there of interest?

**Reply by jeff-h** [reply]
Can you paste in your nginx configuration for Plausible?

**Reply by krchavdarov** [reply]
`
server {
	# replace example.com with your domain name
	server_name stat.az-moga.bg;
	location / {
		proxy_pass http://localhost:8000;
		proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for; 

	}

    listen [::]:443 ssl ipv6only=on; # managed by Certbot
    listen 443 ssl; # managed by Certbot
    ssl_certificate /etc/letsencrypt/live/stat.az-moga.bg/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/stat.az-moga.bg/privkey.pem; # managed by Certbot
    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot

}
server {
    if ($host = stat.az-moga.bg) {
        return 301 https://$host$request_uri;
    } # managed by Certbot


	server_name stat.az-moga.bg;
	
	listen 80;
	listen [::]:80;
    return 404; # managed by Certbot


}
`

**Reply by jeff-h** [reply]
I don't see anything unusual in that nginx config. Indeed, based on the info you've posted above I visited your site (hope you don't feel that is an invasion of your privacy) and it appears to be loading `plausible.js` just fine; I got no CORS errors. Are the errors you initially pasted screenshots of from a different site? They are not incurred when running a development version of your site on localhost perchance?
